odxtools/basicstructure.py

```diff
@@ -31,6 +26,9 @@ class BasicStructure(ComplexDop):
     parameters: NamedItemList[Parameter]
     byte_size: Optional[int]
 
+    def __post_init__(self) -> None:
+        assert isinstance(self, CompositeCodec)
```

Comment:
Why?

Reply:
to somehow enforce that the class adheres to the CompositeCodec protocol. A better way would be to use inheritance, but whenever I tried I ran into troubles with MRO. Deriving a class from a CompositeCodec (which derives from Codec) and a base class that is a Codec but not a CompositeCodec does not work. This happens e.g., for BasicStructure...

Reply:
such checks should be unit tests, not runtime checks.

Reply:
isn't mypy doing the type checking already anyway?

Reply:
good point. fixed: https://github.com/mercedes-benz/odxtools/pull/364/commits/1ebedcea1ef6b3e28a512aa602c0d2478ff542a0